openmls/src/group/mls_group/creation.rs

```diff
@@ -97,6 +97,21 @@ impl MlsGroup {
         Ok(mls_group)
     }
 
+    /// Creates a new group from a [`Welcome`] message. Returns an error
+    /// ([`WelcomeError::NoMatchingKeyPackage`]) if no [`KeyPackage`]
+    /// can be found.
+    pub fn join<KeyStore: OpenMlsKeyStore>(
+        backend: &impl OpenMlsCryptoProvider<KeyStoreProvider = KeyStore>,
+        mls_group_config: &MlsGroupConfig,
+        welcome: MlsMessageIn,
+        ratchet_tree: Option<RatchetTree>,
+    ) -> Result<Self, WelcomeError<KeyStore::Error>> {
+        let welcome = welcome
+            .into_welcome()
+            .ok_or(WelcomeError::MalformedWelcomeMessage)?;
+        Self::new_from_welcome(backend, mls_group_config, welcome, ratchet_tree)
+    }
```

Comment:
Why do we need this in the public API?

Reply:
Because the API as is isn't really usable. All of the API should operate on `MlsMessage`, something like `Welcome`.
I'm fine putting this behind `test-utils` until all of the API is rewritten. But this where it should go.

Reply:
I agree the API needs changes with the new framing and I created #1326 to track that. Let's look at that separately though, because there might be alternatives to this, e.g. by remodeling `MlsMessageIn`.
So you could either put it behind a test flag or use `welcome.into_welcome().unwrap()` in the interop client. In either case please leave a TOFDO for #1326 somewhere.